openml/datasets/dataset.py

```diff
@@ -831,10 +853,6 @@ def retrieve_class_labels(self, target_name: str = "class") -> Union[None, List[
         -------
         list
         """
-        if self.features is None:
```

Comment:
Why can this be dropped?

Reply:
To clarify, I dropped this because with the new lazy loading,` self.feature` cannot be `None`.
AFAIK, it would make no sense to call this function with the intention of `self.feature is None`, because the function requires values for `self.feature`. Hence, I think defaulting to lazy loading is appropriate here.